File: app/src/main/java/com/example/movies/FavoriteMovieViewModel.java
package com.example.movies;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class FavoriteMovieViewModel extends AndroidViewModel {

    private final MovieDao movieDao;

    public FavoriteMovieViewModel(@NonNull Application application) {
        super(application);
        movieDao = MovieDatabase.getInstance(application).movieDao();
    }

    public LiveData<List<Movie>> getAllFavoriteFilms() {
        return movieDao.getAllFavoriteMovies();
    }

    public LiveData<Movie> getFavoriteFilms(int id) {
        return movieDao.getFavoriteMovie(id);
    }

}

File: app/src/main/java/com/example/movies/FavoriteMovieViewModel.kt
package com.example.movies

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.movies.database.MovieDatabase

class FavoriteMovieViewModel(application: Application) : AndroidViewModel(application) {

    private val movieDao = MovieDatabase.getInstance(application).movieDao()
    val getAllFavoriteFilms get() = movieDao.getAllFavoriteMovies()
}